test/asyncio_tests/test_examples.py

```diff
@@ -1503,3 +1504,45 @@ def MongoClient(**kwargs):
         assert isinstance(res["encrypted_unindexed"], Binary)
 
         await client_encryption.close()
+
+
+class MotorAWSLambdaExamples(AsyncIOTestCase):
+    @asyncio_test
+    async def test_shared_client(self):
+        environ.setdefault("MONGODB_URI", "localhost")
+        # Start AWS Lambda Example 1
+        import os
+
+        from motor.motor_asyncio import AsyncIOMotorClient
+
+        client = AsyncIOMotorClient(host=os.environ.get("MONGODB_URI", "localhost"))
+
+        async def async_handler(event, context):
+            return await client.db.command("ping")
+
+        def lambda_handler(event, context):
+            return asyncio.run(async_handler(event, context))
```

Comment:
asyncio.run does not work here:
```python
$ cat example.py 
import asyncio
from motor.motor_asyncio import AsyncIOMotorClient

client = AsyncIOMotorClient()

async def async_handler(event, context):
    return await client.db.command("ping")

def lambda_handler(event, context):
    return asyncio.run(async_handler(event, context))

lambda_handler('event', {})
lambda_handler('event', {})
lambda_handler('event', {})
```
output:
```python
$ python example.py 
Traceback (most recent call last):
  File "example.py", line 13, in <module>
    lambda_handler('event', {})
  File "example.py", line 10, in lambda_handler
    return asyncio.run(async_handler(event, context))
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "example.py", line 7, in async_handler
    return await client.db.command("ping")
  File "/Users/shane/git/motor/motor/metaprogramming.py", line 73, in method
    return framework.run_on_executor(
  File "/Users/shane/git/motor/motor/frameworks/asyncio/__init__.py", line 74, in run_on_executor
    return loop.run_in_executor(_EXECUTOR, functools.partial(fn, *args, **kwargs))
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/asyncio/base_events.py", line 774, in run_in_executor
    self._check_closed()
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/asyncio/base_events.py", line 508, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
```

Reply:
Motor clients bind to the current event loop and need to keep using that same loop. asyncio.run creates, runs, and then closes a loop which is incompatible with this approach. The only way to use asyncio.run would be to also create a new client for each handler call which we don't want to do since clients are expensive and should be reused across calls. You'll need to create one loop and reuse it to synchronize all calls.

Reply:
Your approach suffers from the get_event_loop() deprecation problem:
```python
>>> asyncio.get_event_loop()
<stdin>:1: DeprecationWarning: There is no current event loop
<_UnixSelectorEventLoop running=False closed=False debug=False>
```